examples/language-modeling/run_lora_clm.py

```diff
@@ -444,10 +461,18 @@ def main():
                 use_auth_token=True if model_args.use_auth_token else None,
                 **dataset_args,
             )
-    if data_args.dataset_name == "tatsu-lab/alpaca":
+        for key in raw_datasets:
+            # if alpaca dataset and sql dataset pass as json file, make sure they could work
+            if sorted(raw_datasets[key].features.keys()) == sorted(["input", "output", "instruction"]):
+                data_args.dataset_name = "tatsu-lab/alpaca"
+            if sorted(raw_datasets[key].features.keys()) == sorted(["question", "context", "answer"]):
+                data_args.dataset_name = "b-mc2/sql-create-context"
+
+    if data_args.dataset_name in ["tatsu-lab/alpaca", "b-mc2/sql-create-context"]:
         # Preprocessing the datasets.
+        is_alpaca = data_args.dataset_name == "tatsu-lab/alpaca"
         for key in raw_datasets:
-            prompts = create_prompts(raw_datasets[key])
+            prompts = create_prompts(raw_datasets[key]) if is_alpaca else create_sql_prompts(raw_datasets[key])
```

Comment:
Not a big fan of this as many datasets could get associated to `create_sql_prompts`. I would rather add a `--sql_prompt` arg to this example so that users can decide to inject the SQL prompt or not.